File: frontend/src/app/shared/models/notification.dto.ts
export interface NotificationDto {
  id:            string;
  actorUsername: string;
  actorAvatarUrl: string | null;
  type:          'LIKE' | 'COMMENT' | 'SUBSCRIPTION' | 'ROOM_INVITE' | 'JOIN_REQUEST';
  ownerUsername: string;
  eventId:       string;
  message:       string;
  read:          boolean;
  createdAt:     string;
}

File: backend/src/main/java/com/hikebuddy/notification/dto/NotificationDto.java
package com.hikebuddy.notification.dto;

public record NotificationDto(
        String id,
        String actorUsername,
        String actorAvatarUrl,
        String type,           // "LIKE" | "COMMENT"
        String ownerUsername,
        String eventId,
        String message,
        boolean read,
        String createdAt
) {}
